Week 1 633020384-3 นาย กิตติพจน์ ทิพย์นางรอง

Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello, Python Course!")

Hello, Python Course!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [1]:
Age = 24
GPA = 2.99
Favorite_Color = "Pink"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  24 <class 'int'>
GPA =  2.99 <class 'float'>
Favorite_Color =  Pink <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [ ]:
num1 = 45
num2 = 54
sum = num1 + num2
minus = num1 - num2
multiple = num1 * num2
divide = num1 / num2
print("Sum = ",sum)
print("Minus = ",minus)
print("Multiple = ",multiple)
print("Divide = ",divide)

Sum =  99
Minus =  -9
Multiple =  2430
Divide =  0.8333333333333334


4. String concatenation and f-strings:


In [ ]:
First_name = "Kittipot"
Last_name = "Thipnangrong"
Full_name = First_name + " " + Last_name
print(f"Hello {Full_name}")

Hello Kittipot Thipnangrong


5. User input and type conversion:


In [ ]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 12
type of A is :  <class 'str'>
12  + 5 =  17


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: สามารถจัดการไฟล์งานเอกสาร สามารถบริหารจัดการและวิเคราะห์ข้อมูลหรือแม้กระทั่งประยุกต์ไปใช้ในด้านเครื่อข่ายความปลอดภัยของ Network

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))

13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

ทำไมถึงเป็นtype Float เพราะว่า เมื่อ int คือจำนวนเต็มมาบวกกับ float จำนวนทศนิยม ในภาษา Python จะแปลงให้เป็น float โดยอัตโนมัติ

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.
<class 'str'>

ทำไมถึงเป็นtype String เพราะว่า การใส่ "" ให้เลข 3 0 จะทำให้คอมพิวเตอร์มองว่าเป็นตัวอักษรนั้นเอง

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

ได้ผลลัพธ์: False
<class 'bool'>

ทำไมถึงเป็นtype Boolean เพราะว่า กำหนดให้ is_sunny เป็น true และ is_raining เป็น false

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: error

ทำไมถึงเป็นtype ไม่สามารถบวกกันได้เพราะเนื่องจาก บรรทัดที่ 3 เป็นการรับข้อมูลซึ่งไป string แล้วนำไปบวก ตัวเลข 5 ซึ่งบวกกันไม่ได้